### PR TITLE
Update to mdbook 0.3.

### PR DESCRIPTION
**Pull request by ehuss**
This updates to mdbook 0.3 which uses relative links, which are much better
(the pages mostly work when viewing on GitHub for example).

### CODE REVIEW

**Comment by ehuss** [comment]
r? @Havvy 
I'd like to get this merged relatively soon.  I need to do a coordinated module update on rust-lang/rust to get everything synced up.

This was mostly automated with [my script](https://gist.github.com/ehuss/b2a647eb69aa8f57bc4fb397656dfac1), with a few minor tweaks.

After this merges, I will assist in updating any open PRs.

**Comment by Havvy** [comment]
Due to lack of sleep and my schedule, I'm unable to look at this for about 3 days. You should ask for a different reviewer.

**Comment by ehuss** [comment]
No worries.  I think Centril is traveling, and I haven't seen @matthewjasper around here for a while.  If nobody is around, I'll just have to wait.

**Comment by Centril** [comment]
Yeah, I'm on vacation in 🇮🇹 and will be back on the 20th.

File: src/expressions/range-expr.md [hunk 1]
```diff
@@ -68,11 +68,11 @@ for i in 1..11 {
...
 
-[_Expression_]: expressions.html
+[_Expression_]: ../expressions.md
 
-[std::ops::Range]:            https://doc.rust-lang.org/std/ops/struct.Range.html
-[std::ops::RangeFrom]:        https://doc.rust-lang.org/std/ops/struct.RangeFrom.html
-[std::ops::RangeTo]:          https://doc.rust-lang.org/std/ops/struct.RangeTo.html
-[std::ops::RangeFull]:        https://doc.rust-lang.org/std/ops/struct.RangeFull.html
-[std::ops::RangeInclusive]:   https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html
+[std::ops::Range]: ../           https:/doc.rust-lang.org/std/ops/struct.Range.md
```
Comment (matthewjasper) [inline src/expressions/range-expr.md]: These links are broken

Comment (ehuss) [reply src/expressions/range-expr.md]: Oh my, thanks!  Looks like my script got confused by multiple spaces.

**Comment by matthewjasper** [comment]
Thanks.

**Pull request merged**